### PR TITLE
[CARBONDATA-16] BLOCK distribution in query is not correct in query when number of executors are less than the cluster size.

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**
Fixed : https://issues.apache.org/jira/browse/CARBONDATA-16

### CODE REVIEW

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonQueryRDD.scala [hunk 1]
```diff
@@ -107,8 +108,25 @@ class CarbonQueryRDD[K, V](
       )
       if (blockList.nonEmpty) {
         // group blocks to nodes, tasks
+        var activeNodes = DistributionUtil.getNodeList(sparkContext)
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonQueryRDD.scala]: use nodelist only once.
nodeBlockTaskMapping also should be used only once

Comment (mohammadshahidkhan) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonQueryRDD.scala]: Fixed

**Comment by mohammadshahidkhan** [comment]
fixed review comments
Please review

**Comment by gvramana** [comment]
retest this please

**Pull request merged**